src/go/MONIT/ggus_alerting.go

```diff
@@ -0,0 +1,357 @@
+}
+
+//AlertManager API acceptable JSON Data for GGUS Data
+type amJSON struct {
+	Labels struct {
+		Alertname string `json:"alertname"`
+		Severity  string `json:"severity"`
+		Tag       string `json:"tag"`
+	} `json:"labels"`
+	Annotations struct {
+		TicketID        string `json:"TicketID"`
+		Type            string `json:"Type"`
+		VO              string `json:"VO"`
+		Site            string `json:"Site"`
+		Priority        string `json:"Priority"`
+		ResponsibleUnit string `json:"ResponsibleUnit"`
+		Status          string `json:"Status"`
+		Subject         string `json:"Subject"`
+		Scope           string `json:"Scope"`
+		URL             string `json:"URL"`
+	} `json:"annotations"`
+	StartsAt time.Time `json:"startsAt"`
+	EndsAt   time.Time `json:"endsAt"`
+}
+
+//AlertManager GET API acceptable JSON Data struct for GGUS data
+type getAMJSON struct {
+	Status string `json:"status"`
+	Data   []struct {
+		Labels struct {
+			Alertname string `json:"alertname"`
+			Severity  string `json:"severity"`
+			Tag       string `json:"tag"`
+		} `json:"labels"`
+		Annotations struct {
+			Priority        string `json:"Priority"`
+			ResponsibleUnit string `json:"ResponsibleUnit"`
+			Scope           string `json:"Scope"`
+			Site            string `json:"Site"`
+			Status          string `json:"Status"`
+			Subject         string `json:"Subject"`
+			TicketID        string `json:"TicketID"`
+			Type            string `json:"Type"`
+			VO              string `json:"VO"`
+			URL             string `json:"URL"`
+		} `json:"annotations"`
+		StartsAt     time.Time `json:"startsAt"`
+		EndsAt       time.Time `json:"endsAt"`
+		GeneratorURL string    `json:"generatorURL"`
+		Status       struct {
+			State       string        `json:"state"`
+			SilencedBy  []interface{} `json:"silencedBy"`
+			InhibitedBy []interface{} `json:"inhibitedBy"`
+		} `json:"status"`
+		Receivers   []string `json:"receivers"`
+		Fingerprint string   `json:"fingerprint"`
+	} `json:"data"`
+}
+
+//function for parsing JSON data from GGUS Data
+func (data *ggus) parseJSON(jsondata []byte) {
+	err := json.Unmarshal(jsondata, &data)
+	if err != nil {
+		log.Printf("Unable to parse GGUS JSON data, error: %v\n", err)
+	}
+}
+
+//function for fetching JSON data from GGUS Data
+func fetchJSON(filename string) []byte {
+	file, err := os.Open(filename)
+	if err != nil {
+		log.Printf("Unable to open JSON file, error: %v\n", err)
+	}
+
+	defer file.Close()
+
+	jsonData, err := ioutil.ReadAll(file)
+	if err != nil {
+		log.Printf("Unable to read JSON file, error: %v\n", err)
+	}
+
+	if verbose > 1 {
+		log.Println("GGUS Data: " + string(jsonData))
+	}
+
+	return jsonData
+
+}
+
+//function for converting SSB JSON Data into JSON Data required by AlertManager APIs.
+func convertData(data ggus) []byte {
+
+	var temp amJSON
+	var finalData []amJSON
+
+	exstTkt = make(map[string]int)
+
+	for _, each := range data {
+
+		// Ignoring all tickets whose VO is not "cms"
+		if each.VO != "cms" {
```

Comment:
Make it configurable, such that if we will need to change it later we'll have this flexibility, i.e. we should be able to pass filter to the program, but it should have default value "cms".

Reply:
Fixed this issue.